components/server/src/server.rs

```diff
@@ -1466,11 +1392,19 @@ impl TiKVServer<RaftLogEngine> {
                 self.config.storage.block_cache.shared,
             )),
         );
+        if let Some(e) = engines.raft.as_rocks_engine() {
```

Comment:
How about providing a func called `RegisterConfig` which accepts `cfg_controller` for `ConfiguredRaftEngine` instead of `as_rocks_engine`